Tests/pageObject/noticeToMarine-weekfiledownload.page.ts

```diff
@@ -29,6 +36,31 @@ export default class noticeToMarinerWeekDownload
      const result = this.page.$$eval("[id^='filename']", (options : any[]) => {return options.map( option=>option.textContent.trim())}); 
      return result; 
      }
+
+     public async checkDailyFileDownload()
+     {
+      await this.page.waitForSelector("[id^='filename']");
+      const dailyfileName = this.page.$$eval("[id^='filename']", (options : any[]) => {return options.map( option=>option.textContent.trim())}); 
+        
+      if ((await dailyfileName).length > 0) {
+        const fileName = dailyfileName[0];
+        var dailyPageUrl = await (await (await this.page.$("[id^='download'] > a")).getAttribute('href')).trim().split("&");
+        const newDailyPageUrl = dailyPageUrl[1].replace(/%20/g, " ");
+        return newDailyPageUrl;
+      }
+      else {
+        throw new Error("No download File Found")
+      }
+
+     }
+
+     public async getFirstDailyFileName()
+     {
+      await this.page.waitForSelector("[id^='filename']");
+      const dailyfileName = await this.page.$$eval("[id^='filename']", (options : any[]) => {return options.map( option=>option.textContent.trim())}); 
```

Comment:
Playwright documentation here https://playwright.dev/docs/api/class-page#page-eval-on-selector says that you should use locator.evaluateAll or another locator helper functions.  Is it possible to change this?